### PR TITLE
Handle Inf values in NewtonTrustRegion

### PR DESCRIPTION
**Pull request by andreasnoack**
The will lead to NaNs in the gradient and Hessian which have to be handled explicitly.

### CODE REVIEW

File: test/multivariate/solvers/second_order/newton_trust_region.jl [hunk 1]
```diff
@@ -196,4 +196,9 @@ end
     Optim.solve_tr_subproblem!([0, 1.], [-1000 0; 0. -999], 1e-2, ones(2))
 end
 
+@testset "Handle Inf without erroring" begin
```
Comment (pkofod) [inline test/multivariate/solvers/second_order/newton_trust_region.jl]: Is this test valid, or will it hit the problem we saw earlier with literals?

**Comment by pkofod** [comment]
Seems like a good idea. I'm wondering about that test failure :| Can you verify that the test doesn't produce 0 gradients and therefore doesn't fail?

**Comment by andreasnoack** [comment]
The test works as intended locally

File: test/multivariate/solvers/second_order/newton_trust_region.jl (hunk 1 above)
Comment (andreasnoack) [reply test/multivariate/solvers/second_order/newton_trust_region.jl]: I'm using FD here so it produces NaNs as it should.

**Pull request merged**